cli/core/test/e2e/framework/config_operations.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package framework
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+
+	configapi "github.com/vmware-tanzu/tanzu-framework/cli/runtime/apis/config/v1alpha1"
+)
+
+const (
+	ConfigFileName   = "config.yaml"
+	ConfigNGFileName = "config-ng.yaml"
+	ConfigFileDir    = ".config/tanzu/"
+)
+
+// ConfigOps performs "tanzu config" command operations
+type ConfigOps interface {
+	ConfigSetFeatureFlag(path, value string) error
+	ConfigGetFeatureFlag(path string) (string, error)
+	ConfigUnsetFeature(path string) error
+	ConfigInit() error
+	ConfigServerList() error
+	ConfigServerDelete(serverName string) error
+	DeleteCLIConfigurationFiles() error
+	IsCLIConfigurationFilesExists() bool
+}
+
+// configOps is the implementation of ConfOps interface
+type configOps struct {
+	CmdOps
+}
+
+func NewConfOps() ConfigOps {
+	return &configOps{
+		CmdOps: NewCmdOps(),
+	}
+}
+
+// ConfigSetFeature sets the tanzu config feature flag
+func (co *configOps) ConfigSetFeatureFlag(path, value string) (err error) {
+	confSetCmd := ConfigSet + path + " " + value
+	_, _, err = co.Exec(confSetCmd)
+	return err
+}
+
+// ConfigSetFeature sets the tanzu config feature flag
+func (co *configOps) ConfigGetFeatureFlag(path string) (string, error) {
+	out, _, err := co.Exec(ConfigGet)
+	if err != nil {
+		return "", err
+	}
+
+	var node yaml.Node
+	err = yaml.Unmarshal(out.Bytes(), &node)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to construct yaml node from config get output")
+	}
+	cnf, err := convertNodeToClientConfig(&node)
+	if err != nil {
+		return "", err
+	}
+	featureName := strings.Split(path, ".")[len(strings.Split(path, "."))-1]
+	if cnf != nil && cnf.ClientOptions.Features["global"] != nil {
+		return cnf.ClientOptions.Features["global"][featureName], nil
+	}
```

Comment:
`path` here can be `features.<plugin-name>.<featureName>` so we shouldn't expect that it is `features.global.<featureName>`.

Reply:
Logic has updated to consider any plugin-name